packages/vm/src/modules/file_system_cache.rs

```diff
@@ -108,7 +108,7 @@ impl FileSystemCache {
         let filename = checksum.to_hex();
         let file_path = self.modules_path.join(filename);
 
-        let result = unsafe { Module::deserialize_from_file(engine, &file_path) };
+        let result = unsafe { Module::deserialize_from_file_unchecked(engine, &file_path) };
```

Comment:
The checking here is actually nice. For corruted modules it gives errors instead of crashes.

Can you do this?
- make `unchecked_modules` a cache option that defaults to false
- Benchmark checked vs. unchecked instantiations (see "instantiate from fs" in `packages/vm/benches/main.rs`)

Reply:
Results on my M1 Macbook:
```
Cache/load wasm         time:   [563.50 µs 567.52 µs 571.11 µs]
Cache/load wasm unchecked
                        time:   [559.81 µs 562.01 µs 565.37 µs]
Cache/instantiate from fs
                        time:   [743.75 µs 750.63 µs 758.83 µs]
Cache/instantiate from fs unchecked
                        time:   [752.03 µs 769.98 µs 796.07 µs]
```

Reply:
Great, then we don't even need to expose the option prominently to users and just use the checked version.

Reply:
Can you add the bechmark tests to the benchmarking file? It would be good to expand what we test there since a lot of people often have a lot of opinions on what we are suppoised to optimize.

Reply:
Yes, but then we need to expose the option through the `CacheOptions` to the user, right?

Reply:
Hmm, that would be breaking. I put it in here for now: https://github.com/CosmWasm/cosmwasm/issues/1702

What we can do instead is add `fn set_module_unchecked(&mut self, unchecked: bool)` to `Cache`.

Reply:
Good point, will do it like that 👍

Reply:
We might have to rerun the benchmarks. I just happened to notice that the `_unchecked` functions are [not unchecked under the hood](https://github.com/wasmerio/wasmer/pull/3938).

Reply:
With that patch, the difference is quite significant:
```
Cache/instantiate from fs
                        time:   [756.19 µs 762.76 µs 770.81 µs]
                        change: [-1.7079% -0.3564% +1.0060%] (p = 0.62 > 0.05)
                        No change in performance detected.
Cache/instantiate from fs unchecked
                        time:   [588.42 µs 593.82 µs 600.60 µs]
                        change: [-22.836% -21.487% -20.113%] (p = 0.00 < 0.05)
                        Performance has improved.
```

Reply:
My recommendation is to try and stick with checked deserialization where possible, since it significantly reduces the risk of running into UB. The overhead is usually small enough to use it by default.

I'm not familiar with your requirements, but an in-memory LRU / ARC cache for deserialized modules is a good way to mitigate if the difference is significant for you, at the expense of higher memory usage.

A config setting which defaults to checked deserialization is of course reasonable.